### PR TITLE
Cietro

### PR DESCRIPTION
**Pull request by uwieder**
Added a small tabulated file with a .gz extension to the data folder. Created a button for it and set the CsvFileReader to detect the gz extension and use the appropriate reader.

### CODE REVIEW

File: data/criteo.txt [hunk 1]
```diff
@@ -0,0 +1,1000 @@
+1	5	110		16		1	0	14	7	1		306		62770d79	e21f5d58	afea442f	945c7fcf	38b02748	6fcd6dcb	3580aa21	28808903	46dedfa6	2e027dc1	0c7c4231	95981d1f	00c5ffb7	be4ee537	8a0b74cc	4cdc3efa	d20856aa	b8170bba	9512c20b	c38e2f28	14f65a5d	25b1b089	d7c1fc0b	7caf609c	30436bfc	ed10571d
```
Comment (mbudiu-vmw) [inline data/criteo.txt]: Why do we have four different files with the same data?
Only one seems to be necessary.

File: platform/src/main/java/org/hillview/storage/CsvGzReader.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
+package org.hillview.storage;
```
Comment (mbudiu-vmw) [inline platform/src/main/java/org/hillview/storage/CsvGzReader.java]: It looks like we don't need this class at all.

File: web/src/main/java/org/hillview/InitialObjectTarget.java [hunk 3]
```diff
@@ -136,9 +136,12 @@ void prepareFiles(RpcRequest request, RpcRequestContext context) {
         } else if (which == 4) {
             fileNamePattern = "segmentation.csv";
             schemaFile = "segmentation.schema";
-        } else {
+        } else if (which == 5) {
```
Comment (mbudiu-vmw) [inline web/src/main/java/org/hillview/InitialObjectTarget.java]: please use 4 spaces of indentation

File: web/src/main/webapp/npm-debug.log [hunk 4]
```diff
@@ -0,0 +1,112 @@
+0 info it worked if it ends with ok
```
Comment (mbudiu-vmw) [inline web/src/main/webapp/npm-debug.log]: Please delete this file

File: platform/src/main/java/org/hillview/storage/CsvGzReader.java (hunk 2 above)
Comment (uwieder) [reply platform/src/main/java/org/hillview/storage/CsvGzReader.java]: Correct, I thought I removed it from git, apparently I didn't.

File: web/src/main/java/org/hillview/InitialObjectTarget.java [hunk 5]
```diff
@@ -136,10 +136,14 @@ void prepareFiles(RpcRequest request, RpcRequestContext context) {
...
             schemaFile = "segmentation.schema";
+        } else if (which == 5) {
+            config.separator = '\t';
+            fileNamePattern = "criteoTab.gz";
+            schemaFile = "criteo.schema";
         } else {
             throw new RuntimeException("Unexpected operation " + which);
-        }
-
+		}
```
Comment (mbudiu-vmw) [inline web/src/main/java/org/hillview/InitialObjectTarget.java]: Please don't use tabs. You should configure your editor to always use spaces.
If it is emacs I can send you some lisp code.

**Pull request merged**